packages/js/block-templates/src/utils/register-woo-block-type.ts

```diff
@@ -66,11 +68,32 @@ function getEdit<
 			[ getEvaluationContext, hideConditions ]
 		);
 
+		const disabled: boolean = useSelect(
+			( select: typeof WPSelect ) => {
+				if (
+					! disableConditions ||
+					! Array.isArray( disableConditions )
+				) {
+					return false;
+				}
+
+				const evaluationContext = getEvaluationContext( select );
+
+				return disableConditions.some( ( condition ) =>
+					evaluate( condition.expression, evaluationContext )
+				);
+			},
+			[ getEvaluationContext, disableConditions ]
+		);
+
 		if ( ! edit || shouldHide ) {
 			return null;
 		}
 
-		return createElement( edit, props );
+		return createElement( edit, {
+			...props,
+			attributes: { ...props.attributes, disabled },
```

Comment:
What is there is an existing `disabled` attribute that isn't a boolean? Perhaps we should only add ours if one does not exist already?